### PR TITLE
Get K commands for current active project

### PR DESCRIPTION
**Pull request by jchannon**

### CODE REVIEW

File: tests/OmniSharp.Tests/CurrentProjectFacts.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+using System.Collections.Generic;
+using OmniSharp.Models;
+using Xunit;
```
Comment (nosami) [inline tests/OmniSharp.Tests/CurrentProjectFacts.cs]: usings are all over the place

File: tests/OmniSharp.Tests/CurrentProjectFacts.cs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        {
+            return new Request
+            {
+                Line = 0,
+                Column = 0,
+                FileName = fileName,
+                Buffer = source
+            };
+        }
+
```
Comment (nosami) [inline tests/OmniSharp.Tests/CurrentProjectFacts.cs]: remove this line

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                {
+                    currentProjectName = currentProjectName.Substring(0, currentProjectName.IndexOf("+"));
+                }
+
+                var aspnet5Info = new AspNet5WorkspaceInformation(_aspnet5Context);
+
+                var currentProject = aspnet5Info.Projects.FirstOrDefault(x => x.Name == currentProjectName);
+                if (currentProject != null)
+                {
+                    response.Commands = currentProject.Commands;
```
Comment (nosami) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs]: With an endpoint name like `/project`, I'd be expecting to see the project back.

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        private readonly MSBuildContext _msbuildContext;
+
+        public CurrentProjectController(AspNet5Context aspnet5Context, MSBuildContext msbuildContext, OmnisharpWorkspace workspace)
+        {
+            _aspnet5Context = aspnet5Context;
+            _msbuildContext = msbuildContext;
+            _workspace = workspace;
+        }
+
+
```
Comment (nosami) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs]: remove this line

**Comment by nosami** [comment]
The tests aren't running on mono

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+namespace OmniSharp
+{
+    public class CurrentProjectController
+    {
+        private readonly AspNet5Context _aspnet5Context;
+        private readonly OmnisharpWorkspace _workspace;
+        private readonly MSBuildContext _msbuildContext;
+
+        public CurrentProjectController(AspNet5Context aspnet5Context, MSBuildContext msbuildContext, OmnisharpWorkspace workspace)
```
Comment (davidfowl) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs]: Why isn't this on the same controller as the /projects endpoint? It should also be the same information.

Also don't parse the project name and assume the target framework. We have a structure for that.

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        {
+            _aspnet5Context = aspnet5Context;
+            _msbuildContext = msbuildContext;
+            _workspace = workspace;
+        }
+
+        [HttpPost("currentprojectcommands")]
```
Comment (davidfowl) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs]: Yea, I don't think this controller makes any sense. It makes more sense to do what I said above. Use the controller that returns project information to allow returning individual project information. You can also have something that returns the current project.

Comment (jchannon) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.CurrentProject.cs]: I started to put this under /project to mirror the same response as
/projects with just the current project in the response however I'm not
sure what you mean by `Also don't parse the project name and assume the
target framework. We have a structure for that.` I continued with my
substring approach but then couldn't assign `AspNet5WorkspaceInformation`
projects property once I filtered it out.

I'm not that fussed about this at the moment so please clone this branch
and tidy it up.

**Comment by cubitouch** [comment]
Working on that right now with @davidfowl .

**Pull request closed without merging**